### PR TITLE
fix local name out of order

### PR DESCRIPTION
**Pull request by HerrCai0907**
Local name will be out of order when `funcMappedLocals` has a different order with function's local.
However webassembly spec require the order of local name in custom name section (see https://webassembly.github.io/spec/core/appendix/custom.html#binary-namemap).

So this PR move the mapping logic into loop and add sort logic between mapping and writing.

### CODE REVIEW

File: src/wasm/wasm-binary.cpp [hunk 1]
```diff
@@ -857,28 +857,34 @@ void WasmBinaryWriter::writeNames() {
...
+            }
+            localsWithNames.push_back({indexInBinary, func->getLocalName(indexInFunc)});
           }
         }
+        std::sort(
+          localsWithNames.begin(),
+          localsWithNames.end(),
+          [](std::pair<Index, Name> const& a, std::pair<Index, Name> const& b) {
+            return a.first < b.first;
+          });
```
Comment (MaxGraey) [inline src/wasm/wasm-binary.cpp:883]: Just use simple `std::sort(localsWithNames.begin(), localsWithNames.end())`. Vector elements on basis of first element of pairs.

File: src/wasm/wasm-binary.cpp [hunk 2]
```diff
@@ -1117,9 +1119,10 @@ static void writeBase64VLQ(std::ostream& out, int32_t n) {
     }
     // more VLG digit will follow -- add continuation bit (0x20),
     // base64 codes 'g'..'z', '0'..'9', '+', '/'
-    out << char(digit < 20
-                  ? 'g' + digit
-                  : digit < 30 ? '0' + digit - 20 : digit == 30 ? '+' : '/');
+    out << char(digit < 20    ? 'g' + digit
+                : digit < 30  ? '0' + digit - 20
+                : digit == 30 ? '+'
+                              : '/');
```
Comment (kripken) [inline src/wasm/wasm-binary.cpp:1125]: Did you do this automatically, or manually? `clang-format` errors on CI, so if this was automatic, maybe your version of `clang-format` is much newer or older..?

File: src/wasm/wasm-binary.cpp [hunk 3]
```diff
@@ -857,28 +857,30 @@ void WasmBinaryWriter::writeNames() {
...
+              // the same index, they are not mapped in any way.
+              assert(func->imported());
+              indexInBinary = indexInFunc;
+            }
+            localsWithNames.push_back(
+              {indexInBinary, func->getLocalName(indexInFunc)});
           }
         }
+        std::sort(localsWithNames.begin(), localsWithNames.end());
         assert(localsWithNames.size());
```
Comment (kripken) [inline src/wasm/wasm-binary.cpp:880]:
```suggestion
        assert(localsWithNames.size());
        std::sort(localsWithNames.begin(), localsWithNames.end());
```
Slightly clearer to flip the order here, I think. The assertion is on something from the loop above it, and not related to the sort.

**Review by kripken — COMMENTED** [review]
Thanks for the PR! Looks great aside from the comments below.

File: src/wasm/wasm-binary.cpp (hunk 2 above)
Comment (HerrCai0907) [reply src/wasm/wasm-binary.cpp]: I use clang-format automatically. it may caused by the version or some global setting. I will revert it.

**Review by kripken — APPROVED** [review]
Thanks! lgtm. Will merge once tests pass.

**Pull request merged**